Kamien papier nożyce

Szoker6 opened the pull request:
Sprawdź proszę kod czy nadaje się do zaakceptowania.
Brakuje tylko zatrzymania gry po wykonaniu zadanych rund

Szoker6 committed:
Compressed

Szoker6 committed:
Add file

Szoker6 committed:
Add new div

Szoker6 committed:
Add function match.random

Szoker6 committed:
Add file

Szoker6 committed:
Add style gameResult and showNumberOdRounds

Szoker6 committed:
Add style gameResult and showNumberOdRounds

Szoker6 committed:
Add div for result number

Szoker6 committed:
Add function adding results not yet working and add function that check who won

Szoker6 committed:
Compress

Szoker6 committed:
delete

Szoker6 committed:
Modified function adding results

Szoker6 committed:
Add div endoOfGame

Szoker6 committed:
Add function hoverButtons, function EndGame and if in function askHowManyRounds

Szoker6 commented:
dodałem uniemożliwienie dalszego kontynuowania gry po osiągnięciu wybranych przez użytkownika ilości rund

Szoker6 committed:
Delete empty line

Szoker6 committed:
delete function hoverButtons and updateResults

Szoker6 commented:
Zmodyfikowałem 4 funkcje ktore miały wspolne argumenty; dzieki temu zastosowalem DRY

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,104 @@
+    if (isNaN(numbersOfRounds) ||
+        numbersOfRounds <= 0) {
+            alert('Podaj prawidłową wartość');
+        } else if (Number.isInteger(numbersOfRounds)) {
```
Po co ten case? Prompt nie zwraca liczb tylko stringi

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,104 @@
+        showNumberOfRounds.innerHTML = 'ilość rund: ' + totalRoundsForGame;
+        document.getElementById('stone').style.display = "block";
+        document.getElementById('paper').style.display = "block";
+        document.getElementById('scissors').style.display = "block";      
```
pokazywanie przycisków powinno wylecieć do osobnej funkcji

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,104 @@
+        document.getElementById('scissors').style.display = "block";      
+    });
+    stone.addEventListener('click', function() {
+        choseAnItem('kamień')    
```
nie używamy polskich nazw, a już na pewno nie polskich znaków

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,104 @@
+};
 
+function getRandomChoice() {
+        var randomNumber = (Math.floor(Math.random()*3));
```
brak spacji w tym zapisie

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,104 @@
 
+function showResult(playerChoice, computerChoice) {
+    if ((playerChoice == 'kamień' && computerChoice == 'nożyczki') || (playerChoice == 'papier' && computerChoice == 'kamień') || (playerChoice == 'nożyczki' && computerChoice == 'papier')) {
+        var winValue = parseInt(wins.innerHTML) + 1;
```
nie powinniśmy kalkulować stanu gry na podstawie tego co mamy  w HTMLu - proces powinien być odwrotny. Stan gry trzymaj w JSie, a HTML aktualizuj na jego podstawie

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,104 @@
-function didWon() {
+function endGame(totalRoundsForGame,  gameValue) {
+    if (totalRoundsForGame ===  gameValue) {
+        endOfGame.innerHTML = 'Koniec gry aby rozpoczać od nowa odświerz stronę';
```
to takie nieeleganckie zakończenie żeby wymuszać refresh strony

mentor-kk commented:
do tego zablokuj przycisk nowej gry podczas gdy gra już się toczy

Szoker6 committed:
delete empty line

Szoker6 committed:
Delete empty line

Szoker6 committed:
Add function displayNone and display

Szoker6 commented:
kolejne poprawki

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,122 @@
+winner.innerHTML = wins;
+remixed.innerHTML = draws;
+losser.innerHTML = loss;
+games.innerHTML = game;
```
te 4 linijki powinny uciec do funkcji (np `updateView()`)

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,122 @@
+    var numbersOfRounds = prompt('Ile rund?');  
+    if (isNaN(numbersOfRounds) || numbersOfRounds <= 0) {
+            alert('Podałeś nieprawidłową wartość licznik ustawiony na 5 rund');
+            return numbersOfRounds = 5;       
```
1) wystarczy samo `return '5'` - wiesz dlaczego?
2) To powinien być string 5 nie liczba - wiesz dlaczego?

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,122 @@
+   
+function registerListeners() {
+    beginButton.addEventListener('click', function() {
+        displayNone()
```
nazewnictwo => np `hideStartGameButtons`

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,122 @@
+        var totalRoundsForGameAsString = askHowManyRounds();
+        totalRoundsForGame = Number(totalRoundsForGameAsString);
+        showNumberOfRounds.innerHTML = 'ilość rund: ' + totalRoundsForGame;
+        display();       
```
nazewnictwo => np `showChoiceButtons`

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,122 @@
+function showResult(playerChoice, computerChoice) {
+    if ((playerChoice == 'kamień' && computerChoice == 'nożyczki') || (playerChoice == 'papier' && computerChoice == 'kamień') || (playerChoice == 'nożyczki' && computerChoice == 'papier')) {
+        showWhoWon.innerHTML = 'Wygrana!' + '<br/>' + 'Twój wybór: ' + playerChoice + '<br/>' + 'Wybór komputera: ' + computerChoice + '<br>';     
+        wins = wins + 1 ,winner.innerHTML = wins;
```
pisz takie rzeczy od nowych liniek, nie po przecinku

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,122 @@
+        endOfGame.innerHTML = 'Abt rozpoczać od nowa kliknij nowa gra';
+        stone.style.display = "none";
+        paper.style.display = "none";
+        scissors.style.display = "none";
```
masz funkcje od pokazywania przycisków, to albo zrób analogiczną od ukrywania.. albo najlepiej zrób z niej jedną, bardziej konfigurowalną funkcję, która przyjmie parametr "none" lub "block' i wtedy odpowiednio się zachowa

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,122 @@
 
+function reset() {
+    newGame.addEventListener('click', function() { 
+    wins = loss = game = draws = 0;
```
rozbij - to jest bardzo niebezpieczne w jednej linijce

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,122 @@
 };
 
+function reset() {
+    newGame.addEventListener('click', function() { 
```
ten listener powinien być rejestrowany raz, razem z resztą listenerów - obecnie w apce możliwy jest wyciek (jak odpalisz reset() więcej niż raz)

Szoker6 committed:
Add style for newButon

Szoker6 committed:
Compress

Szoker6 committed:
Add

Szoker6 committed:
Add function updateView, hideStartGameButtons, setButtonsDisplay(display)

newGame Listener moved to other Listener, change number to string for return numberOfRounds

Szoker6 commented:
Wprowadziłem wszystkie poprawki. Proszę o sprawdzenie

mentor-kk commented on index.html:
```diff
@@ -7,25 +7,31 @@
+        <div id="showWhoWon"></div>
+        <div id="showNumberOfRounds"></div>
+        <div>Wygrana Remis Przegrana Ilość rund</div>
+        <div class ="gameresults" id = "wins"></div>
```
te spacje do usunięcia

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,125 @@
+    var numbersOfRounds = prompt('Ile rund?');  
+    if (isNaN(numbersOfRounds) || numbersOfRounds <= 0) {
+            alert('Podałeś nieprawidłową wartość licznik ustawiony na 5 rund');
+            return numbersOfRounds = '5';       
```
po co w tym miejscu `numbersOfRounds ` przypisywać wartość? Pytałem w poprzednim CR czy wiesz dlaczego, ale nie doczekałem się odpowiedzi :P przemyśl jaka to jest różnica i przegadamy na callu jutro

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,125 @@
+    if (totalRoundsForGame ===  game) {
+        endOfGame.innerHTML = 'Abt rozpoczać od nowa kliknij nowa gra';
+        setButtonsDisplay('none');   
+        newGame.style.display = "block";    
```
`newGame` => `newGameButton` będzie jasne, że  to przycisk

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,125 @@
-});
+function endGame(totalRoundsForGame,  game) {
+    if (totalRoundsForGame ===  game) {
+        endOfGame.innerHTML = 'Abt rozpoczać od nowa kliknij nowa gra';
```
`endOfGame` => `endOfGameMessage`

mentor-kk commented on js/script.js:
```diff
@@ -1,61 +1,125 @@
+function hideStartGameButtons(display) {
+    beginButton.style.display = display;   
+    newGame.style.display = display;   
+};
```
zobacz jaka niekonsekwencja, linijka po linijce a zupełnie inna konwencja nazewnictwa zmiennych

Szoker6 committed:
delete space and empty div show buttons

Szoker6 committed:
Compress

Szoker6 committed:
Change id newGame to newGameButton

Szoker6 committed:
deleted

Szoker6 committed:
Change name newGame to newGameButton, change neme endOfGame to endOfGameMessage, add resset counters in function updateView
